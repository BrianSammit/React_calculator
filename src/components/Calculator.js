import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      total: '',
      next: '',
      operation: '',
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const result = calculate(this.state, buttonName);
    this.setState(() => result);
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <>
        <div className="calculator">
          <h1>Letâ€™s do some math!</h1>
          <div>
            <Display total={total} next={next} operation={operation} />
            <ButtonPanel clickHandler={this.handleClick} />
          </div>
        </div>
      </>
    );
  }
}

export default Calculator;
