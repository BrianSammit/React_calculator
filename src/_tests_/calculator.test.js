import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Calculator />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
