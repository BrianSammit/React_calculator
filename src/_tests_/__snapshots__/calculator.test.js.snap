// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="calc"
>
  <h2>
    Let’s do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <h1>
        
        
        
      </h1>
    </div>
    <div
      className="button_panel"
    >
      <div>
        <button
          className="sc-bdnxRM biEjNA button"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          <h3>
            AC
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          <h3>
            +/-
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="%"
          onClick={[Function]}
          type="button"
        >
          <h3>
            %
          </h3>
        </button>
        <button
          className="sc-bdnxRM cVlGdk button"
          name="÷"
          onClick={[Function]}
          type="button"
        >
          <h3>
            ÷
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-bdnxRM biEjNA button"
          name="7"
          onClick={[Function]}
          type="button"
        >
          <h3>
            7
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="8"
          onClick={[Function]}
          type="button"
        >
          <h3>
            8
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="9"
          onClick={[Function]}
          type="button"
        >
          <h3>
            9
          </h3>
        </button>
        <button
          className="sc-bdnxRM cVlGdk button"
          name="X"
          onClick={[Function]}
          type="button"
        >
          <h3>
            X
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-bdnxRM biEjNA button"
          name="4"
          onClick={[Function]}
          type="button"
        >
          <h3>
            4
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="5"
          onClick={[Function]}
          type="button"
        >
          <h3>
            5
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="6"
          onClick={[Function]}
          type="button"
        >
          <h3>
            6
          </h3>
        </button>
        <button
          className="sc-bdnxRM cVlGdk button"
          name="-"
          onClick={[Function]}
          type="button"
        >
          <h3>
            -
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-bdnxRM biEjNA button"
          name="3"
          onClick={[Function]}
          type="button"
        >
          <h3>
            3
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="2"
          onClick={[Function]}
          type="button"
        >
          <h3>
            2
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="1"
          onClick={[Function]}
          type="button"
        >
          <h3>
            1
          </h3>
        </button>
        <button
          className="sc-bdnxRM cVlGdk button"
          name="+"
          onClick={[Function]}
          type="button"
        >
          <h3>
            +
          </h3>
        </button>
      </div>
      <div>
        <button
          className="sc-bdnxRM eYEbbW button_0"
          name="0"
          onClick={[Function]}
          type="button"
        >
          <h3>
            0
          </h3>
        </button>
        <button
          className="sc-bdnxRM biEjNA button"
          name="."
          onClick={[Function]}
          type="button"
        >
          <h3>
            .
          </h3>
        </button>
        <button
          className="sc-bdnxRM cVlGdk button"
          name="="
          onClick={[Function]}
          type="button"
        >
          <h3>
            =
          </h3>
        </button>
      </div>
    </div>
  </div>
</div>
`;
